cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -126,6 +122,135 @@ impl LockedManifest {
         Ok(ContainerBuilder::new(container_builder_path))
     }
 
+    pub fn read_from_file(path: &CanonicalPath) -> Result<Self, LockedManifestError> {
+        let contents = fs::read(path).map_err(LockedManifestError::ReadLockfile)?;
+        serde_json::from_slice(&contents).map_err(LockedManifestError::ParseLockfile)
+    }
+
+    pub fn check_lockfile(
+        path: &CanonicalPath,
+    ) -> Result<Vec<LockfileCheckWarning>, LockedManifestError> {
+        let mut pkgdb_cmd = Command::new(Path::new(&*PKGDB_BIN));
+        pkgdb_cmd
+            .args(["manifest", "check"])
+            .arg("--lockfile")
+            .arg(path.as_os_str());
+
+        debug!("checking lockfile with command: {}", pkgdb_cmd.display());
+
+        let value = call_pkgdb(pkgdb_cmd).map_err(LockedManifestError::CheckLockfile)?;
+        let warnings: Vec<LockfileCheckWarning> =
+            serde_json::from_value(value).map_err(LockedManifestError::ParseCheckWarnings)?;
+
+        Ok(warnings)
+    }
+}
+
+impl ToString for LockedManifest {
+    fn to_string(&self) -> String {
+        serde_json::json!(self).to_string()
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct LockedManifestCatalog {
+    #[serde(rename = "lockfile-version")]
+    version: Version<1>,
+    /// original manifest that was locked
+    manifest: TypedManifestCatalog,
+    /// locked groups
+    groups: Vec<LockedGroup>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct LockedGroup {
+    /// name of the group
+    name: String,
+    /// system this group provides packages for
+    system: System,
+    /// [CatalogPage] that was selected to fulfill this group
+    ///
+    /// If resolution of a group provides multiple pages,
+    /// a single page is selected based on cross group constraints.
+    /// By default this is the latest page that provides packages
+    /// for all requested systems.
+    page: CatalogPage,
+}
+
+impl LockedManifestCatalog {
+    /// Convert a locked manifest to a list of installed packages for a given system
+    /// in a format shared with the pkgdb based locked manifest.
+    pub fn list_packages(&self, system: &System) -> Vec<InstalledPackage> {
+        self.groups
+            .iter()
+            .filter(|group| &group.system == system)
+            .flat_map(|group| {
+                group.page.packages.iter().cloned().map(|package| {
+                    let priority = self
+                        .manifest
+                        .install
+                        .values()
+                        .find(|install| {
+                            install.pkg_path == package.attr_path
+                                && install.package_group.as_deref().unwrap_or("toplevel")
+                                    == group.name
+                        })
+                        .and_then(|install| install.priority)
+                        .unwrap_or(5);
```

Comment:
We may want to lock the priority after all, to avoid `buildenv` using differnet defaults than `flox`.